.github/workflows/pull_request_label_management.yml

```diff
@@ -0,0 +1,141 @@
+name: Pull Request Label Management
+
+on:
+  pull_request_target:
+    types: [labeled, unlabeled]
+
+env:
+  PAT_TOKEN: ${{ secrets.TEAM_READ_PAT }}  # PAT with read:org scopes
+
+permissions: {}
+
+jobs:
+  process-labels:
+    name: Process Label Actions
+    runs-on: ubuntu-22.04
+    permissions:
+      pull-requests: write
+
+    steps:
+      - name: Handle label changes
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ env.PAT_TOKEN }}
+          script: |
+            const { context, github } = require('@actions/github');
```

Comment:
these should already be available per the docs: https://github.com/actions/github-script